### PR TITLE
Don't attempt to rewind the Orchard wallet to a height after its current checkpoint

### LINKED ISSUE #5958: Assertion `uResultHeight == rewindHeight` failed
I was trying to run the smoke tests and I somehow managed to get its wallet into a bad state that triggers the following assertion failure. I have a copy of the `.zcash` directory that reproduces this reliably, but unfortunately I can't reproduce it from scratch. 

```
$ ZCASHD=./src/zcashd ZCASHCLI=./src/zcash-cli ./qa/zcash/smoke_tests.py --wallet=wallet.smoketest.dat "$HOME/.zcash" --automate --use-faucet
Start time: 2022-05-13 11:31:03.438752
Starting zcashd...
zcashd: wallet/wallet.cpp:4413: int CWallet::ScanForWalletTransactions(CBlockIndex *, bool, bool): Assertion `uResultHeight == rewindHeight' failed.
^CTraceback (most recent call last):
  File "/home/firexware/zcash/zcash/./qa/zcash/smoke_tests.py", line 735, in <module>
    main()
  File "/home/firexware/zcash/zcash/./qa/zcash/smoke_tests.py", line 691, in main
    zcash.start()
  File "/home/firexware/zcash/zcash/./qa/zcash/smoke_tests.py", line 610, in start
    subprocess.check_call(cli_args, stdout=devnull)
  File "/usr/lib/python3.10/subprocess.py", line 364, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/usr/lib/python3.10/subprocess.py", line 347, in call
    return p.wait(timeout=timeout)
  File "/usr/lib/python3.10/subprocess.py", line 1204, in wait
    return self._wait(timeout=timeout)
  File "/usr/lib/python3.10/subprocess.py", line 1938, in _wait
    (pid, sts) = self._try_wait(0)
  File "/usr/lib/python3.10/subprocess.py", line 1896, in _try_wait
    (pid, sts) = os.waitpid(self.pid, wait_flags)
KeyboardInterrupt
```

I'm not sure what I did that might have caused it originally, I definitely re-ran and ctrl+c'd the smoke tests quite a bit it's also possible another node tried to access the same .zcash directory or something.

### PR DESCRIPTION
**Pull request by daira**
This fixes #5958.

### CODE REVIEW

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -4695,7 +4695,9 @@ int CWallet::ScanForWalletTransactions(
 
         // no need to read and scan block, if block was created before
         // our wallet birthday (as adjusted for block time variability)
-        while (chainActive.Next(pindex) != NULL && nTimeFirstKey && pindex->GetBlockTime() < nTimeFirstKey - TIMESTAMP_WINDOW) {
+        auto optOrchardCheckpointHeight = orchardWallet.GetLastCheckpointHeight();
```
Comment (nuttycom) [inline src/wallet/wallet.cpp:4698]:
```suggestion
        // only advance the active chain tip as far as the last Orchard checkpoint height
        auto optOrchardCheckpointHeight = orchardWallet.GetLastCheckpointHeight();
```

**Review by nuttycom — DISMISSED** [review]
utACK with suggestion for an additional clarifying comment

**Review by sellout — APPROVED** [review]

**Review by nuttycom — DISMISSED** [review]
utACK

**Comment by sellout** [comment]
utACK

**Review by sellout — APPROVED** [review]
utACK

**Pull request merged**